Source/WTF/wtf/text/AtomString.cpp

```diff
@@ -117,15 +117,10 @@ AtomString AtomString::number(double number)
     return AtomString::fromLatin1(numberToString(number, buffer));
 }
 
-AtomString AtomString::fromUTF8Internal(const char* start, const char* end)
+AtomString AtomString::fromUTF8Internal(std::span<const char> characters)
 {
-    ASSERT(start);
-
-    // Caller needs to handle empty string.
-    ASSERT(!end || end > start);
-    ASSERT(end || start[0]);
-
-    return AtomStringImpl::addUTF8(start, end ? end : start + std::strlen(start));
+    ASSERT(!characters.empty());
```

Comment:
It wasn't previously and I don't think we'd have a security bug in release if the assertion didn't hold. As a result, I'm included to keep as debug to avoid hurting perf in a refactoring patch.